a10_octavia/common/utils.py

```diff
@@ -292,3 +292,14 @@ def validate_interface_vlan_map(hardware_device):
         device_network_map.append(device_map)
     validate_vcs_device_info(device_network_map)
     return device_network_map
+
+
+def verify_acos_version(acos_version):
```

Comment:
For verify_acos_version(), it can return if version is > 5.2. 
In the future if we have other feature need to do similar thing, we need to write another API to verify another specific version. So,
- can we add a constant to specify the feature version.
(for example: ACOS_DYNAMIC_INTERFACE_VERSION = "5.2.0") 
- And provide an API to compare versions.
for example: 
```
if acos_versoin_cmp(vthunder.acos_version, a10constant.ACOS_DYNAMIC_INTERFACE_VERSION) >= 0:
    ....
```

Reply:
>In the future if we have other feature need to do similar thing, we need to write another API to verify another specific version.

We should not take the driver this direction. The AXAPI changes so often, between patches even, that trying to support a heterogeneous device deployment would be impossible. (I know that's not what's being proposed but it sets precedent).

We need to stick to having the acos-client deal with the versioning and enforce a homogenous device deployment. Making this driver API-aware will only lead to problems down the road beyond adding to the already high cognitive load.

For this solution, we should add a `reload_reboot` function to the `actions.py` file of the acos-client library.

```
def reload_reboot(self):
    version_summery = get_acos_version()
    version = version_summary['version']['oper']['sw-version'].split(',')[0]
    if float(version[0:3]) >= 5.2:
         reload()
    else:
         reboot()
```

Reply:
Added the `reload_reboot() `method to the `action.py` file of the `acos-client` library.
It will be risky to check like `if float(acos_version[0:3]) >= 5.2:`  ,as it may get wrong if major/minor number in acos_version > 9.
So the above specified method is modified as follows:

```
def reload_reboot(self, acos_version=None):
        if not acos_version:
            version_summary = self.get_acos_version()
            acos_version = version_summary['version']['oper']['sw-version'].split(',')[0]

        major = acos_version.split('.')[0]
        minor = acos_version.split('.')[1]
        patch = acos_version.split('.')[2]

        if major >= 5 and minor >= 2 and patch >= 0:
            self.probe_network_devices()
            self.reload()
        else:
            self.reboot()
```